Assets/Lua/Doom/dsda-data.lua

```diff
@@ -0,0 +1,1864 @@
+-- Core must be loaded to get memory domain sizes
+-- Throw an error so Lua doesn't cache the module in an invalid state
+assert(emu.getsystemid() == "Doom", "Doom core not loaded")
+
+local dsda = {
+	doom = {},
+	heretic = {},
+	hexen = {},
+}
+
+ -- Read the high word of every pointer and check that it matches the expected base address (for debugging)
+dsda.check_pointers = false
+
+
+
+-- Constants ---
+local NULL_OBJECT <const>    = 0x88888888 -- no object at that index
+local OUT_OF_BOUNDS <const>  = 0xFFFFFFFF -- no such index
+local WBX_POINTER_HI <const> = 0x36F
+local BusDomain <const>      = "System Bus"
+
+dsda.MAX_PLAYERS = 4
+-- sizes in bytes
+dsda.LINE_SIZE   = 256  -- sizeof(line_t) is 232, but we padded it for niceness
+dsda.MOBJ_SIZE   = 512  -- sizeof(mobj_t) is 464, but we padded it for niceness
+dsda.PLAYER_SIZE = 1024 -- sizeof(player_t) is 729, but we padded it for niceness
+dsda.SECTOR_SIZE = 512  -- sizeof(sector_t) is 344, but we padded it for niceness
+
+
+
+-- Locals
+local read_u8   = memory.read_u8
+local read_u24  = memory.read_u24_le
+local read_u32  = memory.read_u32_le
+local readfloat = memory.readfloat
```

Comment:
Similar shortcuts exist in the main script, looks like it makes sense to declare them once and then use everywhere?

Reply:
I don't think it makes sense to share them across scripts, but in any case they're not used anymore in `things-lines.lua` because all the memory reads happen through the dsda script now. 

Unless you would like to keep them for something, I'll just remove them from `things-lines.lua`

Reply:
Okay, please remove them then!